packages/osd-optimizer/src/worker/bundle_refs_plugin.ts

```diff
@@ -134,64 +119,16 @@ export class BundleRefsPlugin {
     });
   }
 
-  private cachedResolveRefEntry(ref: BundleRef) {
-    const cached = this.resolvedRefEntryCache.get(ref);
-
-    if (cached) {
-      return cached;
-    }
-
-    const absoluteRequest = Path.resolve(ref.contextDir, ref.entry);
-    const promise = this.cachedResolveRequest(absoluteRequest).then((resolved) => {
-      if (!resolved) {
-        throw new Error(`Unable to resolve request [${ref.entry}] relative to [${ref.contextDir}]`);
-      }
-
-      return resolved;
-    });
-    this.resolvedRefEntryCache.set(ref, promise);
-    return promise;
-  }
-
-  private cachedResolveRequest(absoluteRequest: string) {
-    const cached = this.resolvedRequestCache.get(absoluteRequest);
-
-    if (cached) {
-      return cached;
-    }
-
-    const promise = this.resolveRequest(absoluteRequest);
-    this.resolvedRequestCache.set(absoluteRequest, promise);
-    return promise;
-  }
-
-  private async resolveRequest(absoluteRequest: string) {
-    const stats = await safeStat(absoluteRequest);
-    if (stats && stats.isFile()) {
-      return absoluteRequest;
-    }
-
-    // look for an index file in directories
-    if (stats?.isDirectory()) {
-      for (const ext of RESOLVE_EXTENSIONS) {
-        const indexPath = Path.resolve(absoluteRequest, `index${ext}`);
-        const indexStats = await safeStat(indexPath);
-        if (indexStats?.isFile()) {
-          return indexPath;
+  private async resolve(request: string, context: string, compiler: webpack.Compiler) {
+    const resolver = compiler.resolverFactory.get('normal');
+    return new Promise<string | undefined>((resolve) => {
+      resolver.resolve({}, context, request, {}, (err: Error, resolvedPath: string) => {
+        if (err) {
+          resolve(undefined);
```

Comment:
Do we want to silently fail here for all resolver errors? I'm not sure how this is handled upstream, but it may simplify build issue debugging if we reject with error so we don't swallow it.

Reply:
The `resolve` function is called within `function maybeReplaceImport()`, even it silently fail here, webpack will still throw error when it resolve the path later. It won't swallow  the error. But I think you have a good point, we can reject here directly, because if the path cannot be resolved here, it cannot be resolved later by webpack either.

I've updated the code, please take a look again, thanks!